### PR TITLE
Add Supabase auth guide

### PR DESCRIPTION
**Pull request by kevinzunigacuellar**
#### Description (required)
Add a Supabase guide to Astro

#### Related issues & labels (optional)

- New content
- hacktoberfest

### CODE REVIEW

**Review by tony-sull — COMMENTED** [review]
Very nicely done, this is an excellent guide!

I've always been a big fan of this setup keeping the Supabase library on the server, exposing API Routes instead of using Supabase (and similar libraries) directly in the client 🙌

File: src/content/docs/en/guides/backend/supabase.mdx [hunk 1]
```diff
@@ -3,11 +3,487 @@ title: Supabase & Astro
...
+
+### Prerequisites
+
+- A Supabase project. If you don't have one, you can sign up for free at [supabase.com](https://supabase.com/) and create a new project.
+- An Astro project with [server-side rendering (SSR)](/en/guides/server-side-rendering/#enabling-ssr-in-your-project) enabled.
+- Supabase credentials for your project. You can find these in the **Settings > API** tab of your Supabase project.
+  - `SUPABASE_URL`: The URL of your Supabase project.
+  - `SUPABASE_ANON_KEY`: The anonymous key for your Supabase project.
+
+### Adding Firebase credentials
```
[the remark this answers is not in the document]
Comment (kevinzunigacuellar) [reply src/content/docs/en/guides/backend/supabase.mdx:22]: Thanks! completely missed that 😅

**Comment by kevinzunigacuellar** [comment]
Note: Supabase has released a new package `@supabase/ssr` for handling SSR clients currently in beta. Once it is stable and their options have been set, happy to migrate the guide to use it.

**Comment by sarah11918** [comment]
@kevinzunigacuellar Can this be closed because of the other PR?

**Comment by kevinzunigacuellar** [comment]
closing in favor of #4371

**Pull request closed without merging**